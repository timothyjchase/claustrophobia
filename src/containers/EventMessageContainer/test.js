import React from 'react'
import { EventMessageContainer, mapStateToProps } from './index'

describe('<EventMessageContainer />', () => {
  describe('with event matching current phase', () => {
    const props = {
      currentState: {
        phase: 'Test Phase',
        upcomingEvent: { phase: 'Test Phase' },
      },
      removeEvent: jest.fn(),
    }
    const component = shallow(<EventMessageContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with event matching not current phase', () => {
    const props = {
      currentState: {
        phase: 'Test Phase 1',
        upcomingEvent: { phase: 'Test Phase 2' },
      },
      removeEvent: jest.fn(),
    }
    const component = shallow(<EventMessageContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with no event', () => {
    const props = {
      currentState: {
        phase: 'Test Phase',
        upcomingEvent: null,
      },
      removeEvent: jest.fn(),
    }
    const component = shallow(<EventMessageContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mapStateToProps', () => {
  describe('with basic state', () => {
    const state = { current: {} }
    it('should return currentState', () => {
      expect(mapStateToProps(state)).toEqual({ currentState: {} })
    })
  })
})
