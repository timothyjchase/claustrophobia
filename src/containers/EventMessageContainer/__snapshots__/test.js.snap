// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventMessageContainer /> with event matching current phase should match snapshot 1`] = `
<EventMessage
  removeEvent={[MockFunction]}
  upcomingEvent={
    Object {
      "phase": "Test Phase",
    }
  }
/>
`;

exports[`<EventMessageContainer /> with event matching not current phase should match snapshot 1`] = `""`;

exports[`<EventMessageContainer /> with no event should match snapshot 1`] = `""`;
